Unused warnings in Mesh with `deprecated=no` (`_convert_old_array` and `_old_primitives`)

### Godot version

27d6bce

### System information

popos 20.04 Linux

### Issue description

Code errors when dev=yes.

```
scene/resources/mesh.cpp:658:14: error: unused function '_convert_old_array' [-Werror,-Wunused-function]
static Array _convert_old_array(const Array &p_old) {
             ^
scene/resources/mesh.cpp:673:28: error: unused variable '_old_primitives' [-Werror,-Wunused-variable]
static Mesh::PrimitiveType _old_primitives[7] = {
```

### Steps to reproduce
```
scons -j`nproc` dev=yes tests=no LINKFLAGS='-Wl,--build-id' use_thin_lto=yes platform=linuxbsd symbols=yes use_lld=yes use_llvm=yes use_mingw=no bits=64 deprecated=no target=release_debug custom_modules=../groups_godot_modules
```

### Minimal reproduction project

Not applicable. This breaks without a project.

--- a/scene/resources/mesh.cpp
+++ b/scene/resources/mesh.cpp
@@ -653,6 +653,7 @@ enum OldArrayFormat {
 
 };
 
+#ifndef DISABLE_DEPRECATED
 static Array _convert_old_array(const Array &p_old) {
 	Array new_array;
 	new_array.resize(Mesh::ARRAY_MAX);
@@ -677,6 +678,7 @@ static Mesh::PrimitiveType _old_primitives[7] = {
 	Mesh::PRIMITIVE_TRIANGLE_STRIP,
 	Mesh::PRIMITIVE_TRIANGLE_STRIP
 };
+#endif // DISABLE_DEPRECATED
 
 void _fix_array_compatibility(const Vector<uint8_t> &p_src, uint32_t p_old_format, uint32_t p_new_format, uint32_t p_elements, Vector<uint8_t> &vertex_data, Vector<uint8_t> &attribute_data, Vector<uint8_t> &skin_data) {
 	uint32_t dst_vertex_stride;

